### ISSUE TITLE
Add option to run without BinaryCache

### ISSUE THREAD

**Issue by univert** [lucasg/Dependencies#131]
BinaryCache is inefficient when running against huge amount of dlls (i.e. large applications) with hundreds of dlls. This is not only disk space overhead also additional I/O overhead and will also cause cache thrashing . So when I am sure that file locking is not an issue, I'd like to run without BinaryCache to gain some flexibility

**Comment by lucasg** [comment]
Yeah, totally agree. However to prevent file locking I need to change completely my PE parser (I currently use ProcessHacker's phlib).

It's in my TODO list, but unfortunately there are a lot of items higher up in the food chain

**Comment by AndhikaWB** [comment]
I also agree to this. Analyzing complex program (such as `Photoshop.exe` in my case) can take up to 500 MB BinaryCache size.

**Comment by lucasg** [comment]
I'm implemented a pretty rudimentary version: you can choose it in the "Options>Preferences>Use BinaryCache"  but the settings is application wide **and you need to restart Dependencies** since BinaryCache is implemented as a singleton.

The default settings is to use BinaryCache. You can test it by downloading artifacts from b19e49d42b5ec3c4ea06e5da516f646f4e0a6459

Without:
![image](https://user-images.githubusercontent.com/2520861/118393239-5a0d9800-b63e-11eb-8f98-7b247c776c9c.png)

With:
![image](https://user-images.githubusercontent.com/2520861/118393227-4bbf7c00-b63e-11eb-90a4-69f8e2f9a5f4.png)

**Comment by techxdave** [comment]
It would be nice to have a command line option for running without the Binary Cache as the cache directory accumulates and becomes quite large for us (order of 100GB).
